### PR TITLE
General attribute docs improvements

### PR DESCRIPTION
**Pull request by Havvy**
Each commit does its own small thing.

This builds on top of #350, so merge that before taking a look at this.

### CODE REVIEW

File: src/items/implementations.md [hunk 1]
```diff
@@ -136,5 +136,11 @@ impl Seq<bool> for u32 {
 }
 ```
 
+## Attributes in Implementations
+
+Implementations may contain inner [attributes] inside the brackets that contain
+the associated items. They must come before the associated items.
```
Comment (matthewjasper) [inline src/items/implementations.md]: Could there be a sentence on what attributes are allowed on impls?

**Comment by Havvy** [comment]
Addressed comments. Also added which attributes are allowed on block expressions. And apparently, as per the example I included on the block expressions page, block expressions also allow outer attributes and the only place you'd be able to tell that isn't overlapping with statements taking attributes is the return expression of a block expression. Go figure.

I'll add which attributes are allowed on functions / modules / the main module/ other items / variants and fields later.

**Comment by Havvy** [comment]
Rebased against master due to merge conflict.

**Comment by Havvy** [comment]
Rebased again for another merge conflict.

**Pull request merged**